core/src/main/java/org/apache/accumulo/core/security/crypto/CryptoModuleFactory.java

```diff
@@ -172,23 +162,16 @@ private static SecretKeyEncryptionStrategy instantiateSecreteKeyEncryptionStrate
     }
 
     if (!implementsSecretKeyStrategy) {
-      log.warn("Configured Accumulo secret key encryption strategy \"%s\" does not implement the SecretKeyEncryptionStrategy interface. No encryption will be used.");
-      return new NullSecretKeyEncryptionStrategy();
+      throw new IllegalArgumentException(
```

Comment:
Again, I'm not sure any of this is necessary if the class is loaded with the ConfigurationTypeHelper from the config. It may be redundant. I'm not 100% sure on this, though.